addon/-private/system/store.js

```diff
@@ -1872,7 +1872,10 @@ Store = Service.extend({
     }
     if (data) {
       // normalize relationship IDs into records
-      this._backburner.schedule('normalizeRelationships', this, this._setupRelationships, internalModel, data);
+      this._pushedInternalModels.push(internalModel, data);
+      if (this._relationshipFlush === null) {
+        this._relationshipFlush = this._backburner.schedule('normalizeRelationships', this, this._setupRelationships);
```

Comment:
rather then forcing the caller to notice this. I would make `_setupRelationships` do this, that way we don't need to leak the scheduling to the caller.